src/lib.rs

```diff
@@ -225,11 +227,11 @@ impl TestConfig {
         let files = find_tests(&self.test_path)?;
         let tests = files.iter()
             .map(|file| parse_test(file, self))
-            .collect::<Vec<_>>();
+            .collect::<TestResult<Vec<_>>>()?;
 
-        let mut failing_tests = 0;
+        let (mut failing_tests, mut total_tests) = (0, 0);
         for test in tests {
-            let test = test?;
+            print!("goldentesting '{}'... ", &test.path.display());
```

Comment:
When running an error-ing test this will output the first line of the error on the same line as "goldentesting xxx....".

Perhaps we should only print once the test succeeds/fails and move this to lines 251/253. For long-running tests cargo will hide the output until the test fails anyway.

Reply:
If it makes it any better, `cargo` also has the same behavior with printing `ok`s, it's kind of unavoidable since you don't know if the test has failed until after you've run it, but you also want to give feedback at the start of the test so that users know something is happening

Reply:
That's true, though since cargo hides the output by default users won't be able to see the feedback until it errors anyway. I suspect somewhat few people use `--no-capture` by default so I'd like to make the output pretty for those who do.

This may be outside the scope of this PR but one solution is collecting the errors for a given test instead of printing them immediately. Then we can print `error:` instead of `ok` and print the errors after.